processing_handler.go

```diff
@@ -60,8 +60,19 @@ func setCacheControl(rw http.ResponseWriter, originHeaders map[string]string) {
 	}
 
 	if len(cacheControl) == 0 && len(expires) == 0 {
-		cacheControl = fmt.Sprintf("max-age=%d, public", config.TTL)
-		expires = time.Now().Add(time.Second * time.Duration(config.TTL)).Format(http.TimeFormat)
+		if val, ok := originHeaders["FallbackTTL"]; ok {
+			FallbackTTL, err := strconv.Atoi(val)
+			if err != nil {
+				cacheControl = fmt.Sprintf("max-age=%d, public", config.TTL)
+				expires = time.Now().Add(time.Second * time.Duration(config.TTL)).Format(http.TimeFormat)
+			} else {
+				cacheControl = fmt.Sprintf("max-age=%d, public", FallbackTTL)
+				expires = time.Now().Add(time.Second * time.Duration(config.FallbackTTL)).Format(http.TimeFormat)
+			}
+		} else {
+			cacheControl = fmt.Sprintf("max-age=%d, public", config.TTL)
+			expires = time.Now().Add(time.Second * time.Duration(config.TTL)).Format(http.TimeFormat)
+		}
```

Comment:
We can DRY this. Set a `ttl` variable to `convig.TTL` value, change it to `config.FallbackTTL` if `FallbackImage` header is presented (value doesn't matter). Then use the `ttl` variable to craft the cache control headers.

Reply:
Hi @DarthSim, the code was rewritten